src/test/java/org/primeframework/mvc/test/RequestBuilder.java

```diff
@@ -832,18 +833,29 @@ HTTPResponseWrapper run() {
 
     // Now that the cookies are ready, if the CSRF token is enabled and the parameter isn't set, we set it to be consistent
     // since the [@control.form] would normally set that into the form and into the request.
-    MVCConfiguration configuration = injector.getInstance(MVCConfiguration.class);
-    if (configuration.csrfEnabled()) {
-      CSRFProvider csrfProvider = injector.getInstance(CSRFProvider.class);
-      if (csrfProvider.getTokenFromRequest(request) == null) {
-        String token = csrfProvider.getToken(request);
-        if (token != null) {
-          String parameterName = csrfProvider.getParameterName();
-          request.addURLParameter(parameterName, token);
+    if (request.getMethod() == HTTPMethod.POST) {
+      MVCConfiguration configuration = injector.getInstance(MVCConfiguration.class);
+      if (configuration.csrfEnabled()) {
+        CSRFProvider csrfProvider = injector.getInstance(CSRFProvider.class);
+        if (csrfProvider.getTokenFromRequest(request) == null) {
+          String token = csrfProvider.getToken(request);
+          if (token != null) {
+            String parameterName = csrfProvider.getParameterName();
+            requestBodyParameters.put(parameterName, List.of(token));
+          }
         }
       }
     }
 
+    // Optionally allow the caller to consume the HTTP request in order to mutate it and what not.
+    try {
+      RequestBuilderHTTPRequestConsumer httpRequestConsumer = injector.getInstance(RequestBuilderHTTPRequestConsumer.class);
+      if (httpRequestConsumer != null) {
+        httpRequestConsumer.accept(request);
+      }
+    } catch (ConfigurationException ignore) {
+    }
```

Comment:
This is a little non-standard in that because this is an optional binding I am just catching the exception. 
Anyone have any concerns with this pattern?